dev/io.openliberty.security.fips.internal_fat/fat/src/io/openliberty/security/fips/fat/tests/fips1403/security/utility/FIPS1403SecurityUtilityTests.java

```diff
@@ -280,8 +295,61 @@ public void fips140_3CreateLTPAKeysTest() throws Exception {
 
     }
 
+    @Test
+    @AllowedFFDC({ "java.lang.RuntimeException" })
+    public void fips140_3NoSuchAlgorithmExceptionTest() throws Exception {
+
+        // Enable FIPS at server level without referencing the custom profile
+        ProgramOutput po = runSecurityUtilityCommand(new String[] {SEC_CONF_FIPS_COMMAND, OPT_SERVER + "=" + SERVER_NAME});
+        assertEquals("securityUtility configureFIPS did not result in expected return code.",0, po.getReturnCode());
+        
+        server.startServer();
+        ShrinkHelper.defaultDropinApp(server, "Sha1App.war", "com.example.sha1");
+        assertNotNull("Application Sha1App did not start", server.waitForStringInLog("CWWKZ0001I.*Sha1App"));
+        
+        // Access the SHA-1 servlet to trigger the NoSuchAlgorithmException
+        String urlString = "http://" + server.getHostname() + ":" + server.getHttpDefaultPort() + "/Sha1App";
+        URL url = new URL(urlString);
+
+        HttpURLConnection con = HttpUtils.getHttpConnection(url, HttpURLConnection.HTTP_INTERNAL_ERROR, 10);
+        Log.info(thisClass, "fips140_3NoSuchAlgorithmExceptionTest", "HTTP Response Code: " + con.getResponseCode());
+        assertEquals("Expected HTTP 500 Internal Server Error when FIPS blocks SHA-1", HttpURLConnection.HTTP_INTERNAL_ERROR, con.getResponseCode());
+        assertNotNull("Expected NoSuchAlgorithmException for SHA-1 algorithm in server logs when FIPS is enabled", server.waitForStringInLog("NoSuchAlgorithmException.*SHA-1"));
+        
+        // Shutdown server ignoring NoSuchAlgorithmExceptions in server logs and disable FIPS
+        server.stopServer("SRVE0777E", "SRVE0315E");
+        disableFIPS(FIPS_TARGET.SERVER, SERVER_NAME);
+
+        // Copy the SHA1-Application.properties file to the server's resources/security directory
+        Log.info(thisClass, "fips140_3NoSuchAlgorithmExceptionTest", "Applying custom profile to allow SHA-1 for SHA1Servlet");
+        String sha1CustomProfilePath = server.getServerRoot() + "/resources/security/" + CUSTOM_SHA1_FIPS_PROFILE_FILENAME;
+        server.copyFileToLibertyServerRoot("publish/resources", "resources/security", CUSTOM_SHA1_FIPS_PROFILE_FILENAME);
+        File sha1CustomProfileFile = new File(sha1CustomProfilePath);
```

Comment:
one more suggestion; instead of copying over a file, can we edit the file that was created earlier and just append the SHA-1 constraint instead?

this would be more similar to how a user would do it and we wouldn't have to maintain that SHA-1 file if the template file changes.